### PR TITLE
enable machinePool settings and nodeReplace ssh support

### PR DESCRIPTION
**Pull request by slickwarren**
supports tests in: https://github.com/rancher/rancher/pull/43753

### CODE REVIEW

**Review by susesgartner — APPROVED** [review]
Looks pretty good. Ship it!

**Review by markusewalker — APPROVED** [review]

**Review by igomez06 — COMMENTED** [review]
overall looks good, just one comment.

**Review by caliskanugur — COMMENTED** [review]
I've one nit, looks good to me

File: extensions/nodes/node_status.go [hunk 1]
```diff
@@ -142,3 +146,41 @@ func IsNodeReplaced(client *rancher.Client, oldMachineID string, clusterID strin
...
+// Otherwise, an error is returned.
+func Isv1NodeConditionMet(client *rancher.Client, machineID, clusterID, conditionReason string) error {
+	steveclient, err := client.Steve.ProxyDownstream(clusterID)
+	if err != nil {
+		return err
+	}
+
+	v1NodeStatus := &rkev1.RKEMachineStatus{}
+
+	err = wait.Poll(PollInterval, PollTimeout, func() (done bool, err error) {
```
[the remark this answers is not in the document]
Comment (slickwarren) [reply extensions/nodes/node_status.go]: went ahead and updated the entire file's polls

**Review by igomez06 — APPROVED** [review]
LGTM. Can we get the back ports and the JJB PRs ready, before we merge?

File: extensions/nodes/node_status.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (caliskanugur) [reply extensions/nodes/node_status.go]: Thank you @slickwarren

**Review by caliskanugur — APPROVED** [review]
Looks good to me

**Review by markusewalker — APPROVED** [review]
LGTM

**Pull request merged**